### PR TITLE
En_Rail_Skb

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Rail_Skb/z_en_rail_skb.c [hunk 1]
```diff
@@ -101,122 +158,1003 @@ static DamageTable D_80B7346C = {
...
+    this->unk_3FA = 0;
+
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+            break;
+
+        case 3:
+            if ((globalCtx->gameplayFrames & 1) != 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Rail_Skb/z_en_rail_skb.c:526]: Would this match with a `% 2`
```suggestion
            if ((globalCtx->gameplayFrames % 2) != 0) {
```

File: src/overlays/actors/ovl_En_Rail_Skb/z_en_rail_skb.c [hunk 2]
```diff
@@ -101,122 +158,1003 @@ static DamageTable D_80B7346C = {
...
+        func_80B717C8(this);
+        this->unk_3F2 = 1;
+    }
+
+    if (func_80147624(globalCtx)) {
+        switch (this->unk_400) {
+            case 5100:
+                func_801518B0(globalCtx, 5101, &this->actor);
+                this->unk_400 = 5101;
+                break;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Rail_Skb/z_en_rail_skb.c:711]: Those are `textId`s. We use hex for them.
```suggestion
        switch (this->unk_400) {
            case 0x13EC:
                func_801518B0(globalCtx, 0x13ED, &this->actor);
                this->unk_400 = 0x13ED;
                break;
```
Please change every use of `unk_400` accordingly

**Review by AngheloAlf — COMMENTED** [review]
A few comments

**Review by engineer124 — APPROVED** [review]
LGTM, as long as the other comments are addressed

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**